src/Context/UserInfo.tsx

```diff
@@ -0,0 +1,122 @@
+import { createContext, useContext, useReducer } from "react";
+
+// Define Type for UserContextProvider
```

Comment:
You should avoid adding comments over each element. The goal is to write code that is clear enough to be self-explanatory. If you need to add a comment is usually because there is additional information necessary: https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments/
```suggestion
```